docs/tutorial/installation.md

```diff
@@ -43,22 +43,75 @@ npm install --platform=win32 electron
 
 If you need to use an HTTP proxy you can [set these environment variables](https://github.com/request/request/tree/f0c4ec061141051988d1216c24936ad2e7d5c45d#controlling-proxy-behaviour-using-environment-variables).
 
+## Custom Mirrors and Caches
+During installation, the `electron` module will call out to [`electron-download`](https://github.com/electron-userland/electron-download) to download prebuilt
+binaries of Electron for your platform. It will do so by contacting GitHub's
+release download page (https://github.com/electron/electron/releases/download/v).
+
+If you are unable to access GitHub or you need to provide a custom build, you
+can do so by either providing mirror or by providing an existing cache directory.
+
+#### Mirror
+You can override the base url, the path at which to look for Electron binaries,
+and the binary filename using environment variables. The url used by `electron-download`
+is composed as follows:
+
+```
+url = ELECTRON_MIRROR + ELECTRON_CUSTOM_DIR + '/' + ELECTRON_CUSTOM_FILENAME
+```
+
+For instance, to use the China mirror:
+
+```
+ELECTRON_MIRROR="https://npm.taobao.org/mirrors/electron/"
+```
+
+#### Cache
+Alternatively, you can override the local cache. `electron-download` will cache
+downloaded binaries in a local directory to not stress your network. You can use
+that cache folder to provide custom builds of Electron or to avoid making contact
+with the network at all.
+
+* Linux: `$XDG_CACHE_HOME` or `~/.cache/electron/`
+* MacOS: `~/Library/Caches/electron/`
+* Windows: `$LOCALAPPDATA/electron/Cache` or `~/AppData/Local/electron/Cache/`
+
+On environments that have been using older versions of Electron, you might find the
+cache also in ``~/.electron`.
+
+You can also override the local cache location by providing a `ELECTRON_CACHE`
+environment variable.
+
+The cache contains the version's official zip file as well as a checksum, stored as
+a text file. A typical cache might look like this:
+
+```
```

Comment:
This fenced code block needs a language declaration.. I believe `sh` is our de facto declaration (vs `shell` or `bash`)